### ISSUE TITLE
spurious unused_parens warning

### ISSUE THREAD

**Issue by programmerjake** [rust-lang/rust#55342]
The following code produces a unused_parens warning however if the parenthesis around the `mut v` are removed the code doesn't compile:
```rust
pub fn f(v: &u32) -> u32 {
    let &(mut v) = v;
    v += 1;
    v
}
```

**Comment by estebank** [comment]
This is caused by [the stabilization of parentheses in patterns](https://github.com/rust-lang/rust/issues/51087).
